File: Python/Ugly Number.py
class Solution(object):
    def isUgly(self, num):
        """
        :type num: int
        :rtype: bool
        """
        for p in 2, 3, 5:
            while num % p == 0 and num > 0:
                num /= p
        return num == 1
                

File: cpp/Ugly Number.cpp
class Solution {
public:
    bool isUgly(int num) {
        for (int fact : vector<int> {2,3,5}) {
            while (num > 0 && num % fact == 0) {
                num /= fact;
            }
        }
        return num == 1;
    }
};
